pkg/condition/condition.go

```diff
@@ -0,0 +1,85 @@
+package condition
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+
+	"github.com/go-logr/logr"
+	operators "github.com/operator-framework/api/pkg/operators/v2"
+	"github.com/pkg/errors"
+	core "k8s.io/api/core/v1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubeTypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/openshift/windows-machine-config-operator/pkg/patch"
+)
+
+const (
+	UpgradeableFalseMessage = "upgradeIsNotSafe"
+	UpgradeableFalseReason  = "The operator is currently processing sub-components."
+	UpgradeableTrueMessage  = "upgradeIsSafe"
+	UpgradeableTrueReason   = "The operator is safe for upgrade."
+
+	// OperatorConditionName is an environment variable set by OLM identifying the operator's OperatorCondition CR
+	OperatorConditionName = "OPERATOR_CONDITION_NAME"
+)
+
+// ToggleOLMUpgrade sets the current operator Upgradeable condition to either prevent or allow operator version upgrades
+// Sends warning event and logs error in case of failure. No-op if operator is not managed by OLM
+func ToggleOLMUpgrade(c client.Client, recorder record.EventRecorder, log logr.Logger, ocNamespace string, allow bool) {
```

Comment:
I am not a fan of exposing such a method. I prefer having explicit helpers like enable and disable. This makes reading the code much easier at the call site.